### DISCUSSION TITLE
How do I use sequences in sqlalchemy? (in postgres and oracle?)

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [sqlalchemy/sqlalchemy#8241] [Usage Questions]
How do I use sequences in sqlalchemy? (in postgres and oracle?)
Is there a way to use them without model? 
I need to get next_val(), do some logic and only then insert it in a table

**Comment by CaselIT** [comment]
Hi,

Have you tried searching for Sequence in the docs? that would point you to https://docs.sqlalchemy.org/en/14/core/defaults.html?highlight=sequence#sqlalchemy.schema.Sequence
Since 1.4 you can also use identity columns by using Identity

**Reply by martinenkoEduard** [reply]
I managed to complete it like that -
  seq = db.Sequence('testseq')
  nextid = db.session.execute(seq)
